src/providers.rs

```diff
@@ -1,406 +1,302 @@
-pub const CLOUDFLARE_HOSTNAME: &str = "cloudflare-eth.com";
-pub const BLOCKPI_ETH_MAINNET_HOSTNAME: &str = "ethereum.blockpi.network";
-pub const BLOCKPI_ETH_SEPOLIA_HOSTNAME: &str = "ethereum-sepolia.blockpi.network";
-pub const PUBLICNODE_ETH_MAINNET_HOSTNAME: &str = "ethereum-rpc.publicnode.com";
-pub const PUBLICNODE_ETH_SEPOLIA_HOSTNAME: &str = "ethereum-sepolia-rpc.publicnode.com";
-pub const ETH_SEPOLIA_HOSTNAME: &str = "rpc.sepolia.org";
-pub const ALCHEMY_ARBITRUM_ONE_HOSTNAME: &str = "arb-mainnet.g.alchemy.com";
-pub const BLOCKPI_ARBITRUM_ONE_HOSTNAME: &str = "arbitrum.blockpi.network";
-pub const PUBLICNODE_ARBITRUM_ONE_HOSTNAME: &str = "arbitrum-one-rpc.publicnode.com";
-pub const ALCHEMY_BASE_MAINNET_HOSTNAME: &str = "base-mainnet.g.alchemy.com";
-pub const BLOCKPI_BASE_MAINNET_HOSTNAME: &str = "base.blockpi.network";
-pub const PUBLICNODE_BASE_MAINNET_HOSTNAME: &str = "base-rpc.publicnode.com";
-pub const ALCHEMY_OPT_MAINNET_HOSTNAME: &str = "opt-mainnet.g.alchemy.com";
-pub const BLOCKPI_OPTIMISM_MAINNET_HOSTNAME: &str = "optimism.blockpi.network";
-pub const PUBLICNODE_OPTIMISM_MAINNET_HOSTNAME: &str = "optimism-rpc.publicnode.com";
-pub const LLAMA_ETH_MAINNET_HOSTNAME: &str = "eth.llamarpc.com";
-pub const LLAMA_ARBITRUM_ONE_HOSTNAME: &str = "arbitrum.llamarpc.com";
-pub const LLAMA_BASE_MAINNET_HOSTNAME: &str = "base.llamarpc.com";
-pub const LLAMA_OPTIMISM_MAINNET_HOSTNAME: &str = "optimism.llamarpc.com";
-
-// Limited API credentials for local testing.
-// Use `dfx canister call evm_rpc updateProvider ...` to pass your own keys.
-pub const ALCHEMY_ETH_MAINNET_CREDENTIAL: &str = "/v2/zBxaSBUMfuH8XnA-uLIWeXfCx1T8ItkM";
-pub const ALCHEMY_ETH_SEPOLIA_CREDENTIAL: &str = "/v2/Mbow19DWsfPXiTpdgvRu4HQq63iYycU-";
-pub const ALCHEMY_ARBITRUM_ONE_CREDENTIAL: &str = "/v2";
-pub const ALCHEMY_BASE_MAINNET_CREDENTIAL: &str = "/v2";
-pub const ALCHEMY_OPTIMISM_MAINNET_CREDENTIAL: &str = "/v2";
-pub const BLOCKPI_ETH_MAINNET_CREDENTIAL: &str = "/v1/rpc/0edc81e20be23ddff051f61a97bb457ec7284a58";
-pub const BLOCKPI_ETH_SEPOLIA_CREDENTIAL: &str = "/v1/rpc/1fe987fddded17db50862311720ff444991d4dab";
-pub const BLOCKPI_ARBITRUM_ONE_CREDENTIAL: &str =
-    "/v1/rpc/a8b89a41d2a341e32ee7aefcb20820a7cbb65f35";
-pub const BLOCKPI_BASE_MAINNET_CREDENTIAL: &str =
-    "/v1/rpc/bd458bf9f28ed45c77823814a937c812d2efd260";
-pub const BLOCKPI_OPTIMISM_MAINNET_CREDENTIAL: &str =
-    "/v1/rpc/d54bfe59299d56b0cbb8b3c69bd122f4ab5ac654";
-
-pub fn get_default_providers() -> Vec<RegisterProviderArgs> {
-    vec![
-        RegisterProviderArgs {
-            chain_id: ETH_MAINNET_CHAIN_ID,
-            hostname: CLOUDFLARE_HOSTNAME.to_string(),
-            credential_path: "/v1/mainnet".to_string(),
-            credential_headers: None,
-            cycles_per_call: 0,
-            cycles_per_message_byte: 0,
+pub const PROVIDERS: &[Provider] = &[
+    Provider {
+        provider_id: 0,
```

Comment:
How would you define "stable" in this context? As a starting point, I added a test to enforce the expected `provider_id` sequence (starting at zero, increasing by one). This covers the cases of removing or skipping a provider, while other changes are unlikely to occur by accident because each ID is hard-coded.